mne/io/tag.py

```diff
@@ -203,6 +204,257 @@ def _loc_to_eeg_loc(loc):
+            sparse_ptrs = np.fromstring(fid.read(4 * (nrow + 1)), dtype='>i4')
+            data = sparse.csr_matrix((sparse_data, sparse_indices,
+                                     sparse_ptrs), shape=shape)
+    else:
+        raise Exception('Cannot handle other than dense or sparse '
+                        'matrices yet')
+    return data
+
+
+def _read_simple(fid, tag, shape, rlims, dtype):
+    return _fromstring_rows(fid, tag.size, dtype=dtype, shape=shape,
+                            rlims=rlims)
+
+
+def _read_string(fid, tag, shape, rlims):
+    # Always decode to unicode.
+    d = _fromstring_rows(fid, tag.size, dtype='>c', shape=shape, rlims=rlims)
+    return text_type(d.tostring().decode('utf-8', 'ignore'))
+
+
+def _read_complex_float(fid, tag, shape, rlims):
+    # data gets stored twice as large
+    if shape is not None:
+        shape = (shape[0], shape[1] * 2)
+    d = _fromstring_rows(fid, tag.size, dtype=">f4", shape=shape, rlims=rlims)
+    d = d[::2] + 1j * d[1::2]
+    return d
+
+
+def _read_complex_double(fid, tag, shape, rlims):
+    # data gets stored twice as large
+    if shape is not None:
+        shape = (shape[0], shape[1] * 2)
+    d = _fromstring_rows(fid, tag.size, dtype=">f8", shape=shape, rlims=rlims)
+    d = d[::2] + 1j * d[1::2]
+    return d
+
+
+def _read_id_struct(fid, tag, shape, rlims):
+    data = dict()
+    data['version'] = int(np.fromstring(fid.read(4), dtype=">i4"))
+    data['machid'] = np.fromstring(fid.read(8), dtype=">i4")
+    data['secs'] = int(np.fromstring(fid.read(4), dtype=">i4"))
+    data['usecs'] = int(np.fromstring(fid.read(4), dtype=">i4"))
+    return data
+
+
+def _read_dig_point_struct(fid, tag, shape, rlims):
+    data = dict()
+    data['kind'] = int(np.fromstring(fid.read(4), dtype=">i4"))
+    data['ident'] = int(np.fromstring(fid.read(4), dtype=">i4"))
+    data['r'] = np.fromstring(fid.read(12), dtype=">f4")
+    data['coord_frame'] = FIFF.FIFFV_COORD_UNKNOWN
+    return data
+
+
+def _read_coord_trans_struct(fid, tag, shape, rlims):
+    from ..transforms import Transform
+    fro = int(np.fromstring(fid.read(4), dtype=">i4"))
+    to = int(np.fromstring(fid.read(4), dtype=">i4"))
+    rot = np.fromstring(fid.read(36), dtype=">f4").reshape(3, 3)
+    move = np.fromstring(fid.read(12), dtype=">f4")
+    trans = np.r_[np.c_[rot, move],
+                  np.array([[0], [0], [0], [1]]).T]
+    data = Transform(fro, to, trans)
+    fid.seek(48, 1)  # Skip over the inverse transformation
+    return data
+
+
+def _read_ch_info_struct(fid, tag, shape, rlims):
+    d = dict()
+    d['scanno'] = int(np.fromstring(fid.read(4), dtype=">i4"))
+    d['logno'] = int(np.fromstring(fid.read(4), dtype=">i4"))
+    d['kind'] = int(np.fromstring(fid.read(4), dtype=">i4"))
+    d['range'] = float(np.fromstring(fid.read(4), dtype=">f4"))
+    d['cal'] = float(np.fromstring(fid.read(4), dtype=">f4"))
+    d['coil_type'] = int(np.fromstring(fid.read(4), dtype=">i4"))
```

Comment:
@jona-sassenhagen I tried implementing this but it ends up looking a bit clunky since sometimes we cast, other times we don't. So I'm going to go with @dengemann's idea.